src/api/models.py

```
@@ -16,21 +17,52 @@ class Spot(models.Model):
+        if self.status == 'cancelled':
+            return 'Booking is already cancelled.'
+        
+        if self.end_at <= timezone.now():
+            return 'Booking time has already completed and cannot be cancelled.'
+            
+        return None
 
     class Meta:
         indexes = [
             models.Index(fields=["spot", "start_at", "end_at", "status"])
         ]
```

Ruff (0.14.1)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)